napari/_vispy/layers/labels.py

```diff
@@ -68,8 +73,12 @@
     // we get a different value:
     // - if it's the empty key, exit;
     // - otherwise, it's a hash collision: continue searching
-    while ((abs(found - t) > 1e-8) && (abs(found - empty) > 1e-8)) {
+    float initial_t = t;
+    while ((abs(found - initial_t) > 1e-8) && (abs(found - empty) > 1e-8)) {
         t = t + 1;
+        if (t - initial_t > LUT_shape.x * LUT_shape.y) {
+            return vec4(0);
+        }
```

Comment:
Good catch, I remember my gpu hanging a few times when working on this, probably because of this.